core/Pos/Core/Arbitrary.hs

```diff
@@ -320,26 +354,42 @@ instance Arbitrary G.StakeDistribution where
       , return G.ExponentialStakes
       , G.ExplicitStakes <$> arbitrary
       ]
+    shrink = genericShrink
 
 ----------------------------------------------------------------------------
 -- Arbitrary miscellaneous types
 ----------------------------------------------------------------------------
 
 instance Arbitrary Millisecond where
     arbitrary = fromMicroseconds <$> choose (0, 600 * 1000 * 1000)
+    shrink = shrinkIntegral
 
 instance Arbitrary Microsecond where
     arbitrary = fromMicroseconds <$> choose (0, 600 * 1000 * 1000)
+    shrink = shrinkIntegral
 
 deriving instance Arbitrary Types.Timestamp
 
 newtype SmallHashMap =
     SmallHashMap (HashMap PublicKey (HashMap PublicKey (AsBinary Share)))
-    deriving Show
+    deriving (Show, Generic)
 
 instance Arbitrary SmallHashMap where
     arbitrary = SmallHashMap <$> makeSmall arbitrary
+    shrink = genericShrink
 
-derive makeArbitrary ''UnsignedVarInt
-derive makeArbitrary ''SignedVarInt
-derive makeArbitrary ''FixedSizeInt
+instance (Arbitrary a, Integral a) => Arbitrary (UnsignedVarInt a) where
+    arbitrary = genericArbitrary
+    shrink = genericShrink
+
+instance (Arbitrary a, Integral a) => Arbitrary (SignedVarInt a) where
+    arbitrary = genericArbitrary
+    shrink = genericShrink
+
+instance (Arbitrary a, Integral a) => Arbitrary (FixedSizeInt a) where
+    arbitrary = genericArbitrary
+    shrink = genericShrink
```

Comment:
So you added `Integral` constraint, but didn't change it to `shrinkIntegral`, it seems strange.